portfolio/src/main/webapp/index.html

```diff
@@ -82,11 +82,23 @@ <h2>Some Image Transformations!</h2>
           </li>
         </ul>
       </div>
-      <h2>Let's fetch data!</h2>
+      <h4>Join the Discussion!</h4>
       <div class="speech-bubble card">
-        <button onclick="showComments()">Show Comments!</button>
-        <div id="commentsContainer"></div>
+        <form action="/data" method="POST">
+          <textarea name="text-input" placeholder="Leave a message..." id="commentsTextArea"></textarea>
+          <input type="submit" id="commentsSubmitButton"/>
+        </form>
+      </div>
+      <h4 id="commentsCount">Previous comments (0)</h4>
+      <div class="speech-bubble card">
+        <ul id="commentsContainer"></ul>
       </div>
```

Comment:
nit: Instead of showing `Previous comments (0)` before the initial load, maybe don't show anything, or show some sort of loading indicator?

Reply:
That makes sense, I think I'll go with a loading indicator.